integration/lifecycle/interop_test.go

```diff
@@ -209,9 +209,11 @@ var _ = Describe("Release interoperability", func() {
 			}
 			nwo.DeployChaincode(network, "testchannel", orderer, chaincode)
 
-			By("committing the old transaction, expecting to hit an MVCC conflict")
+			By("committing the old transaction")
+			// FAB-17458: because the endorsed tx doesn't have _lifecycle in read set,
+			// it has no read conflict with the cc upgrade via new lifecycle.
 			err = CommitTx(network, env, endorsers[0], deliveryClient, ordererClient, userSigner, txid)
-			Expect(err).To(MatchError(ContainSubstring("transaction invalidated with status (MVCC_READ_CONFLICT)")))
+			Expect(err).NotTo(HaveOccurred())
```

Comment:
In this test, a tx was endorsed in V1_4_2 capability which didn't have _lifecycle in read set. Then the app capability was upgraded to V2_0 and chaincode was upgraded using new lifecycle. Due to no read conflict (the old tx was using lscc and the cc upgrade was using _lifecycle), the old tx was validated and committed. 
Talked to @jyellick, this behavior is acceptable, but open for discussion if there is any concern.

Reply:
One concern is that an inflight tx on old business logic may be committed after the cc is already upgraded with new business logic. 
To keep the existing behavior (MVCC_READ_CONFLICT), one idea is to add 'lscc' to the read set when upgrading cc using new lifecycle. However, it is not preferred because we want to get rid off 'lscc'.

Reply:
@ale-linux worked on the transition from old to new lifecycle, so would be good to get his opinion as well.

Reply:
We discussed further and concluded as Jason did, that the behavior is acceptable. If application really wants to ensure that the latest chaincode is used, they should maintain their own version key in the user chaincode state, read it for every transaction, and increment it upon deploying the new chaincode. This way, if a transaction endorsed at prior version attempts to be committed with a new chaincode already in place, it will fail MVCC validation check. This would work on old lifecycle, new lifecycle, as well as when upgrading from old chaincode to new chaincode.